### PR TITLE
(EAI-581): Chatbot verified answers eval/improvments + export `mongodb-rag-core/dataSources`

### PR DESCRIPTION
**Pull request by mongodben**
Jira: https://jira.mongodb.org/browse/EAI-581

## Changes

- Eval suite for MongoDB Edu chatbot verified answers
- Export data source-related code as separate `/dateSources` named export
  - Done to fix issue in https://github.com/braintrustdata/braintrust-sdk/issues/431
- Increased verified answers `minScore` to `.96` to resolve issue in [EAI-580](https://jira.mongodb.org/browse/EAI-580).
- Add 2 new verified answers to capture eval failures

## Notes

- This eval currently hangs due to issue noted in https://github.com/braintrustdata/braintrust-sdk/issues/433
- Evals https://www.braintrust.dev/app/mdb-test/p/mongodb-chatbot-verified-answers/experiments/mongodb-chatbot-latest-docs-chatbot-embedding-ada-002-minScore-0.96-80352bd6?c=mongodb-chatbot-latest-docs-chatbot-embedding-ada-002-minScore-0.96-6da3c7f8

### CODE REVIEW

File: packages/chatbot-server-mongodb-public/tsconfig.build.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
   "extends": "./tsconfig.json",
-  "exclude": ["**/*.test.ts", "**/*.eval.ts"]
+  "exclude": ["**/*.test.ts"]
```
Comment (nlarew) [inline packages/chatbot-server-mongodb-public/tsconfig.build.json:3]: Does this mean eval files are included in the build?

Comment (mongodben) [reply packages/chatbot-server-mongodb-public/tsconfig.build.json:3]: oh that should def be added back. must have done that during debugging. nice catch.

File: packages/chatbot-server-mongodb-public/src/verifiedAnswers.eval.ts [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    // No similar verified answer
+    tags: ["billing", "should_not_match"],
+    verifiedAnswerIndex,
+  }),
+  makeVerifiedAnswerEvalCase({
+    inputQuery: "how do I set up billing alerts in Atlas?",
+    // No similar verified answer
+    tags: ["billing", "should_not_match"],
+    verifiedAnswerIndex,
+  }),
```
Comment (nlarew) [inline packages/chatbot-server-mongodb-public/src/verifiedAnswers.eval.ts:145]:
```suggestion
```

Is this an unintended dupe of the previous entry?

File: packages/chatbot-server-mongodb-public/src/verifiedAnswers.eval.ts [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+  let matchType = "";
+  if (isMatch && args.expected.answer !== undefined) {
+    matchType = "true_positive";
+  } else if (isMatch && args.expected.answer === undefined) {
+    matchType = "true_negative";
+  } else if (!isMatch && args.expected.answer !== undefined) {
+    matchType = "false_positive";
+  } else if (!isMatch && args.expected.answer === undefined) {
+    matchType = "false_negative";
+  }
```
Comment (nlarew) [inline packages/chatbot-server-mongodb-public/src/verifiedAnswers.eval.ts:195]: Could be a little more concise here

```suggestion
  const matchType = [
    isMatch ? "true" : "false",
    args.expected.answer === undefined ? "negative" : "positive",
  ].join("_");
```

File: packages/mongodb-rag-core/src/index.ts [hunk 4]
```diff
@@ -6,8 +6,8 @@ export * from "./contentStore";
 export * as ContentStore from "./contentStore";
 export * from "./conversations";
 export * as Conversations from "./conversations";
-export * from "./dataSources";
-export * as DataSources from "./dataSources";
+// export * from "./dataSources";
+// export * as DataSources from "./dataSources";
```
Comment (nlarew) [inline packages/mongodb-rag-core/src/index.ts:10]:
```suggestion
```

Should we remove or uncomment?

**Review by nlarew — APPROVED** [review]
LGTM w/ a couple small suggestions

File: packages/chatbot-server-mongodb-public/src/verifiedAnswers.eval.ts (hunk 2 above)
Comment (mongodben) [reply packages/chatbot-server-mongodb-public/src/verifiedAnswers.eval.ts:145]: yea looks like it, nice catch

**Pull request merged**